### PR TITLE
Add support for AzureDNSZone enabled storage accounts used for deep storage

### PR DESCRIPTION
**Pull request by zachjsh**
### Description

Added a new config to AzureAccountConfig

`storageAccountEndpointSuffix`

which allows the user to specify a storage account endpoint suffix where the underlying storage account is enabled for AzureDNSZone. The previous config `endpointSuffix`, did not allow support for such accounts. The previous config has been deprecated in favor of this new config. Also fixed an issue where `managedIdentityClientId` was not being set properly

This PR has:

- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [x] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [x] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: docs/development/extensions-core/azure.md [hunk 1]
```diff
@@ -42,6 +42,6 @@ To use this Apache Druid extension, [include](../../configuration/extensions.md#
 |`druid.azure.protocol`|the protocol to use|http or https|https|
 |`druid.azure.maxTries`|Number of tries before canceling an Azure operation.| |3|
 |`druid.azure.maxListingLength`|maximum number of input files matching a given prefix to retrieve at a time| |1024|
-|`druid.azure.endpointSuffix`|The endpoint suffix to use. Override the default value to connect to [Azure Government](https://learn.microsoft.com/en-us/azure/azure-government/documentation-government-get-started-connect-to-storage#getting-started-with-storage-api).|Examples: `core.windows.net`, `core.usgovcloudapi.net`|`core.windows.net`|
-
+|`druid.azure.endpointSuffix`| The endpoint suffix to use. Override the default value to connect to [Azure Government](https://learn.microsoft.com/en-us/azure/azure-government/documentation-government-get-started-connect-to-storage#getting-started-with-storage-api). This config will be deprecated in a future release in favor of `druid.azure.storageAccountEndpointSuffix`. | Examples: `core.windows.net`, `core.usgovcloudapi.net`                          | Optional                                                                         |
```
Comment (georgew5656) [inline docs/development/extensions-core/azure.md:45]: can you keep the current whitespace for these changes? hard to tell whats going on

File: extensions-core/azure-extensions/src/main/java/org/apache/druid/storage/azure/AzureAccountConfig.java [hunk 2]
```diff
@@ -124,18 +140,76 @@ public void setSharedAccessStorageToken(String sharedAccessStorageToken)
...
 
+  public String getStorageAccountEndpointSuffix()
+  {
+    return storageAccountEndpointSuffix;
+  }
+
   public String getBlobStorageEndpoint()
   {
-    return "blob." + endpointSuffix;
+    if (endpointSuffix != null) {
```
Comment (georgew5656) [inline extensions-core/azure-extensions/src/main/java/org/apache/druid/storage/azure/AzureAccountConfig.java:168]: can you add a comment here explaining this is here to support the legacy runtime property?

File: docs/development/extensions-core/azure.md (hunk 1 above)
Comment (georgew5656) [inline docs/development/extensions-core/azure.md:45]: i would probably just remove this line from the docs if the property is deprecated

Comment (zachjsh) [reply docs/development/extensions-core/azure.md]: fixed

Comment (zachjsh) [reply docs/development/extensions-core/azure.md]: removed

**Review by georgew5656 — APPROVED** [review]

**Review by jon-wei — APPROVED** [review]

**Pull request merged**